### PR TITLE
Expand metrics, add tests.

### PR DESCRIPTION
**Pull request by jkodroff**
Sample output:

```bash
➜  schema-tools git:(jkodroff/add-tests) ./schema-tools stats -p aws
Provider: aws
{
  "Functions": {
    "TotalFunctions": 410,
    "TotalDescriptionBytes": 1225515,
    "TotalInputPropertyDescriptionBytes": 86831,
    "InputPropertiesMissingDescriptions": 32,
    "TotalOutputPropertyDescriptionBytes": 174618,
    "OutputPropertiesMissingDescriptions": 1264
  },
  "Resources": {
    "TotalResources": 1043,
    "TotalDescriptionBytes": 8011210,
    "TotalInputProperties": 10642,
    "InputPropertiesMissingDescriptions": 373,
    "TotalOutputProperties": 11687,
    "OutputPropertiesMissingDescriptions": 572
  }
}%
```

```bash
➜  schema-tools git:(jkodroff/add-tests) ./schema-tools compare -p aws -o v5.9.0 -n v5.10.0
Found 52 breaking changes:
Resource "aws:gamelift/gameServerGroup:GameServerGroup" missing input "tagsAll"
Resource "aws:keyspaces/keyspace:Keyspace" missing input "tagsAll"
Resource "aws:connect/userHierarchyGroup:UserHierarchyGroup" missing input "tagsAll"
Resource "aws:fms/policy:Policy" missing input "tagsAll"
Resource "aws:ec2/defaultVpcDhcpOptions:DefaultVpcDhcpOptions" missing input "netbiosNameServers"
Resource "aws:ec2/defaultVpcDhcpOptions:DefaultVpcDhcpOptions" missing input "netbiosNodeType"
Resource "aws:ec2/defaultVpcDhcpOptions:DefaultVpcDhcpOptions" output "netbiosNameServers" type changed from "array" to "string"
had no type but now has <nil>
Resource "aws:networkmanager/globalNetwork:GlobalNetwork" missing input "tagsAll"
Resource "aws:ec2transitgateway/multicastDomain:MulticastDomain" missing input "tagsAll"
Resource "aws:redshift/hsmClientCertificate:HsmClientCertificate" missing input "tagsAll"
Resource "aws:opensearch/domain:Domain" missing input "tagsAll"
Resource "aws:networkmanager/device:Device" missing input "tagsAll"
Resource "aws:location/placeIndex:PlaceIndex" missing input "tagsAll"
Resource "aws:grafana/workspace:Workspace" missing input "tagsAll"
Resource "aws:kendra/index:Index" missing input "tagsAll"
Resource "aws:ec2transitgateway/connectPeer:ConnectPeer" missing input "tagsAll"
Resource "aws:docdb/eventSubscription:EventSubscription" missing input "tagsAll"
Resource "aws:ec2/networkInsightsPath:NetworkInsightsPath" missing input "tagsAll"
Resource "aws:redshift/usageLimit:UsageLimit" missing input "tagsAll"
Resource "aws:rds/snapshotCopy:SnapshotCopy" missing input "tagsAll"
Resource "aws:networkmanager/site:Site" missing input "tagsAll"
Resource "aws:qldb/stream:Stream" missing input "tagsAll"
Resource "aws:emrcontainers/virtualCluster:VirtualCluster" missing input "tagsAll"
Resource "aws:appflow/flow:Flow" missing input "tagsAll"
Resource "aws:athena/dataCatalog:DataCatalog" missing input "tagsAll"
Resource "aws:iot/provisioningTemplate:ProvisioningTemplate" missing input "tagsAll"
Resource "aws:datasync/fsxOpenZfsFileSystem:FsxOpenZfsFileSystem" missing input "tagsAll"
Resource "aws:lightsail/containerService:ContainerService" missing input "tagsAll"
Resource "aws:location/map:Map" missing input "tagsAll"
Resource "aws:keyspaces/table:Table" missing input "tagsAll"
Resource "aws:emrserverless/application:Application" missing input "tagsAll"
Resource "aws:networkmanager/link:Link" missing input "tagsAll"
Resource "aws:ec2transitgateway/connect:Connect" missing input "tagsAll"
Resource "aws:costexplorer/anomalyMonitor:AnomalyMonitor" missing input "tagsAll"
Resource "aws:cloudtrail/eventDataStore:EventDataStore" missing input "tagsAll"
Resource "aws:redshift/hsmConfiguration:HsmConfiguration" missing input "tagsAll"
Resource "aws:transfer/workflow:Workflow" missing input "tagsAll"
Resource "aws:appconfig/eventIntegration:EventIntegration" missing input "tagsAll"
Resource "aws:connect/routingProfile:RoutingProfile" missing input "tagsAll"
Resource "aws:networkmanager/connection:Connection" missing input "tagsAll"
Function "aws:ec2/getInstanceType:getInstanceType" missing input "gpuses"
Function "aws:ec2/getInstanceType:getInstanceType" missing input "hypervisor"
Function "aws:ec2/getInstanceType:getInstanceType" missing input "inferenceAccelerators"
Function "aws:ec2/getInstanceType:getInstanceType" missing input "instanceDisks"
Function "aws:ec2/getInstanceType:getInstanceType" missing input "totalGpuMemory"
Function "aws:ec2/getInstanceType:getInstanceType" missing input "defaultThreadsPerCore"
Function "aws:ec2/getInstanceType:getInstanceType" missing input "fpgas"
Function "aws:ec2/getInstanceType:getInstanceType" missing input "totalInstanceStorage"
Function "aws:ec2/getInstanceType:getInstanceType" missing input "totalFpgaMemory"
Function "aws:ec2/getInstanceType:getInstanceType" missing input "defaultCores"
Function "aws:ec2/getInstanceType:getInstanceType" missing input "maximumIpv6AddressesPerInterface"

#### New resources:

- `accessanalyzer/archiveRule.ArchiveRule`
- `applicationinsights/application.Application`
- `apprunner/observabilityConfiguration.ObservabilityConfiguration`
- `cognito/riskConfiguration.RiskConfiguration`
- `costexplorer/anomalySubscription.AnomalySubscription`
- `costexplorer/costAllocationTag.CostAllocationTag`
- `directoryservice/sharedDirectory.SharedDirectory`
- `directoryservice/sharedDirectoryAccepter.SharedDirectoryAccepter`
- `kendra/experience.Experience`
- `kendra/faq.Faq`
- `kendra/querySuggestionsBlockList.QuerySuggestionsBlockList`
- `kendra/thesaurus.Thesaurus`
- `lakeformation/lfTag.LfTag`
- `lakeformation/resourceLfTags.ResourceLfTags`
- `lightsail/database.Database`
- `location/routeCalculation.RouteCalculation`
- `location/tracker.Tracker`
- `rum/appMonitor.AppMonitor`
- `transcribe/medicalVocabulary.MedicalVocabulary`

#### New functions:

- `kendra/getExperience.getExperience`
- `kendra/getFaq.getFaq`
- `kendra/getIndex.getIndex`
- `kendra/getQuerySuggestionsBlockList.getQuerySuggestionsBlockList`
- `kendra/getThesaurus.getThesaurus`
- `location/getRouteCalculator.getRouteCalculator`
- `location/getTracker.getTracker`
- `outposts/getAsset.getAsset`
- `outposts/getAssets.getAssets`
- `secretsmanager/getRandomPassword.getRandomPassword`
- `servicediscovery/getHttpNamespace.getHttpNamespace`
- `servicediscovery/getService.getService`
```

Open to making this output a little nicer, but didn't want to get too much scope creep on a project that was initially intended to be confined to Hack Week.

Note that the refactoring to `pkg` and `cmd` will also allow us to de-duplicate this code in the bridge: https://github.com/pulumi/pulumi-terraform-bridge/blob/master/pkg/tfgen/metrics.go#L53-L133

Definite next step should be to integrate this into our CI processes and also pin to a version so we have a repo under the Pulumi org's control and can safely roll out future changes.

Possible future changes:

* The command interface will need breaking changes in the future to accommodate local files and URIs for `stats` and `compare`.
* Unified output flag for json, YAML, MD (will make output on GH issues easier to read), and text, similar to AWS CLI.
* Additional potentially useful stats.

### CODE REVIEW

File: pkg/stats.go [hunk 1]
```diff
@@ -8,44 +8,123 @@ import (
...
-		return totalProperties, propertiesMissingDesc
+
+		res.totalOutputs = len(t.ObjectTypeSpec.Properties)
+
+		for _, output := range t.ObjectTypeSpec.Properties {
+			if output.Description == "" {
+				res.outputsMissingDesc++
+			}
+
+			// TODO: Determine if outputs linking to a type is an actual schema capability
```
Comment (AaronFriel) [inline pkg/stats.go:115]: Yeah, complex output types are possible.

File: pkg/stats.go [hunk 2]
```diff
@@ -8,44 +8,123 @@ import (
...
 			}
 
-			if p.Ref != "" {
-				tn := strings.TrimPrefix(p.Ref, "#/types/")
-				nestedTotalProps, nestedPropsMissingDesc := propCount(tn)
-				totalProperties += nestedTotalProps
-				propertiesMissingDesc += nestedPropsMissingDesc
+			if input.Ref != "" {
+				tn := strings.TrimPrefix(input.Ref, "#/types/")
+				nestedRes := propCount(tn)
```
Comment (AaronFriel) [inline pkg/stats.go:99]: The lookup into `sch.Types[tn]` on line 88 could fail if the `.Ref` typename is in an external package. That doesn't happen often, but it matters for component providers.

Might want to check here:

```go
if _, has := sch.Types[typeName]; has {				 
  nestedRes := propCount(tn)
  ...
}
```

File: pkg/stats.go [hunk 3]
```diff
@@ -54,23 +133,65 @@ func CountStats(sch schema.PackageSpec) PulumiSchemaStats {
...
 			}
 
-			if p.Ref != "" {
-				typeName := strings.TrimPrefix(p.Ref, "#/types/")
-				nestedTotalProps, nestedPropsMissingDesc := propCount(typeName)
-				stats.TotalResourceInputs += nestedTotalProps
-				stats.ResourceInputsMissingDesc += nestedPropsMissingDesc
+			if input.Ref != "" {
+				typeName := strings.TrimPrefix(input.Ref, "#/types/")
+				res := propCount(typeName)
```
Comment (AaronFriel) [inline pkg/stats.go:147]: Same here.

File: pkg/stats.go [hunk 4]
```diff
@@ -54,23 +133,65 @@ func CountStats(sch schema.PackageSpec) PulumiSchemaStats {
...
+		stats.Resources.TotalOutputProperties += len(r.ObjectTypeSpec.Properties)
+
+		for _, output := range r.ObjectTypeSpec.Properties {
+			if output.Description == "" {
+				stats.Resources.OutputPropertiesMissingDescriptions++
+			}
+
+			if output.Ref != "" {
+				typeName := strings.TrimPrefix(output.Ref, "#/types/")
+				res := propCount(typeName)
```
Comment (AaronFriel) [inline pkg/stats.go:164]: And here.

File: pkg/stats.go [hunk 5]
```diff
@@ -8,44 +8,123 @@ import (
...
+
+		for _, output := range t.ObjectTypeSpec.Properties {
+			if output.Description == "" {
+				res.outputsMissingDesc++
+			}
+
+			// TODO: Determine if outputs linking to a type is an actual schema capability
+			if output.Ref != "" {
+				tn := strings.TrimPrefix(output.Ref, "#/types/")
+				nestedRes := propCount(tn)
```
Comment (AaronFriel) [inline pkg/stats.go:118]: etc.

Basically, `.Ref` types could be anywhere, or even one of our "well-known" types like `pulumi.json#/Asset`

File: pkg/stats.go (hunk 2 above)
Comment (jkodroff) [reply pkg/stats.go:99]: Added a test to ensure this does not panic. @AaronFriel Please resolve the convo if the test correctly addresses this concern, or provide additional feedback if not.

File: pkg/stats.go (hunk 1 above)
Comment (jkodroff) [reply pkg/stats.go]: Added a test case that covers this.

**Review by dixler — APPROVED** [review]
LGTM

**Pull request merged**